File: libraries/chr/test_isnam.f
      SUBROUTINE TEST_ISNAM(STATUS)
*+
*  Name:
*     TEST_ISNAM

*  Purpose:
*     Test CHR_ISNAM.

*  Language:
*     Starlink Fortran 77

*  Invocation:
*     CALL TEST_ISNAM(STATUS)

*  Description:
*     Test CHR_ISNAM.
*     If any failure occurs, return STATUS = SAI__ERROR.
*     Otherwise, STATUS is unchanged.

*  Arguments:
*     STATUS = INTEGER (Returned)
*        The status of the tests. 

*  Authors:
*     RLVAD::AJC: A J Chipperfield (STARLINK)
*     RLVAD::ACC: A C Charles (STARLINK)
*     {enter_new_authors_here}

*  History:
*     17-AUG-1989 (RLVAD::AJC):
*        Original version.
*     14-SEP-1993 (ACC)
*        Modularised version: broken into one routine for each of 5 main 
*        categories of tests.
*     02-MAR-1994 (ACC)
*        Second modularised version: broken further into one routine for 
*        each of subroutine tested.  This subroutine created.
*     {enter_further_changes_here}

*  Bugs:
*     {note_any_bugs_here}

*  Subprograms called:    
*     CHR_ISNAM

*-

*  Type Definitions:
      IMPLICIT NONE              ! No implicit typing

*  Arguments Given:
*     None

*  Arguments Returned:
      INTEGER STATUS

*  Global Constants:
      INCLUDE 'SAE_PAR'          ! Standard SAE constants
      INCLUDE 'CHR_ERR'

*  External References:
      LOGICAL CHR_ISNAM

*.

*    Test CHR_ISNAM

      IF (CHR_ISNAM('NAME01') .AND. .NOT.
     : (CHR_ISNAM('01NAME') .OR. CHR_ISNAM('NAME@'))) THEN
         PRINT *, 'CHR_ISNAM OK'
      ELSE
         PRINT *, 'CHR_ISNAM FAILS'
         STATUS = SAI__ERROR
      ENDIF

      END

File: libraries/chr/testIsnam.c
#include <stdio.h>
#include "sae_par.h"
#include "chr.h"

void testIsnam( int *status ){
/*
*+
*  Name:
*     testIsnam

*  Purpose:
*     Test "chrIsnam".

*  Synopsis:
*     void testIsnam( int *status )

*  Description:
*     Test "chrIsnam". If any failure occurs, return "status" = SAI__ERROR.
*     Otherwise, "status" is unchanged.

*  Parameters:
*     *status
*        Returned holding the status of the tests.

*  Copyright:
*     Copyright (C) 2018 East Asian Observatory
*     All rights reserved.

*  Licence:
*     This program is free software; you can redistribute it and/or modify
*     it under the terms of the GNU General Public License as published by
*     the Free Software Foundation; either version 2 of the License, or (at
*     your option) any later version.
*
*     This program is distributed in the hope that it will be useful,but
*     WITHOUT ANY WARRANTY; without even the implied warranty of
*     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
*     General Public License for more details.
*
*     You should have received a copy of the GNU General Public License
*     along with this program; if not, write to the Free Software
*     Foundation, Inc., 51 Franklin Street,Fifth Floor, Boston, MA
*     02110-1301, USA

*  Authors:
*     DSB: David S. Berry (EAO)

*  History:
*     15-MAY-2018 (DSB):
*        Original version, based on equivalent Fortran function.
*-
*/

   if( *status != SAI__OK ) return;

   if( !chrIsnam( "NAME01" ) || chrIsnam( "01NAME" ) || chrIsnam( "NAME@" ) ){
      printf("chrIsnam fails\n");
      *status = SAI__ERROR;
   }

}

